### PR TITLE
Update default 4bit configs

### PR DESCRIPTION
**Pull request by KodiaqQ**
# What does this PR do?

Contains configuration updates based on the experiments from 135227 and following PR's:

- https://github.com/openvinotoolkit/openvino.genai/pull/377
- https://github.com/openvinotoolkit/openvino.genai/pull/419

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by KodiaqQ** [comment]
cc @AlexKoff88

**Comment by echarlaix** [comment]
To fix the code style test you can do the following : 
```
pip install .[quality]
make style
```

**Review by echarlaix — APPROVED** [review]
LGTM, thanks @KodiaqQ

File: optimum/intel/openvino/configuration.py [hunk 1]
```diff
@@ -33,25 +33,33 @@
...
-    "databricks/dolly-v2-3b": {"bits": 4, "sym": False, "group_size": 32, "ratio": 0.5},
+    "databricks/dolly-v2-3b": {"bits": 4, "sym": False, "group_size": 128, "ratio": 0.8},
     "EleutherAI/gpt-j-6b": {"bits": 4, "sym": False, "group_size": 64},
     "facebook/opt-6.7b": {"bits": 4, "sym": False, "group_size": 64, "ratio": 0.8},
     "bigscience/bloomz-7b1": {"bits": 4, "sym": False, "group_size": 32, "ratio": 0.6},
     "togethercomputer/RedPajama-INCITE-7B-Instruct": {"bits": 4, "sym": False, "group_size": 128},
-    "HuggingFaceH4/zephyr-7b-beta": {"bits": 4, "sym": True, "group_size": 64, "ratio": 0.6},
-    "meta-llama/Llama-2-7b": {"bits": 4, "sym": True, "group_size": 128, "ratio": 0.6},
+    "HuggingFaceH4/zephyr-7b-beta": {"bits": 4, "sym": True, "group_size": 128, "ratio": 0.8, 
+                                     "dataset": "wikitext2", "awq": True},
```
Comment (echarlaix) [inline optimum/intel/openvino/configuration.py:42]:
```suggestion
    "HuggingFaceH4/zephyr-7b-beta": {
        "bits": 4,
        "sym": True,
        "group_size": 128,
        "ratio": 0.8,
        "dataset": "wikitext2",
        "awq": True,
    },
```

File: optimum/intel/openvino/configuration.py [hunk 2]
```diff
@@ -33,25 +33,33 @@
...
     "tiiuae/falcon-7b": {"bits": 4, "sym": True, "group_size": 64, "all_layers": True},
     "psmathur/orca_mini_3b": {"bits": 4, "sym": True, "group_size": 64, "all_layers": True},
     "mistralai/Mixtral-8x7B-v0.1": {"bits": 4, "sym": True, "group_size": 128, "ratio": 0.8},
+    "facebook/opt-2.7b": {"bits": 4, "sym": True, "group_size": 128, "ratio": 0.7},
+    "togethercomputer/RedPajama-INCITE-Chat-3B-v1": {"bits": 4, "sym": False, "group_size": 128, "ratio": 0.8},
+    "lmsys/vicuna-7b-v1.5": {"bits": 4, "sym": False, "group_size": 128, "ratio": 1.0},
+    "stabilityai/stablelm-tuned-alpha-3b": {"bits": 4, "sym": False, "group_size": 128, "ratio": 0.8},
+    "mistralai/Mistral-7B-v0.1": {"bits": 4, "sym": True, "group_size": 128, "ratio": 0.9},
+    "baichuan-inc/Baichuan2-7B-Chat": {"bits": 4, "sym": True, "group_size": 128, "ratio": 0.8, 
+                                       "dataset": "wikitext2", "awq": True},
```
Comment (echarlaix) [inline optimum/intel/openvino/configuration.py:62]:
```suggestion
    "baichuan-inc/Baichuan2-7B-Chat": {
        "bits": 4,
        "sym": True,
        "group_size": 128,
        "ratio": 0.8,
        "dataset": "wikitext2",
        "awq": True,
    },
```

File: optimum/intel/openvino/configuration.py [hunk 3]
```diff
@@ -33,25 +33,45 @@
...
+        "dataset": "wikitext2",
+        "awq": True,
+    },
+    "meta-llama/Llama-2-7b": {"bits": 4, "sym": True, "group_size": 128, "ratio": 0.7},
     "meta-llama/Llama-2-7b-chat": {"bits": 4, "sym": True, "group_size": 128, "ratio": 0.8},
     "meta-llama/Llama-2-13b-chat": {"bits": 4, "sym": True, "group_size": 64, "ratio": 0.8},
     "stabilityai/stablelm-3b-4e1t": {"bits": 4, "sym": True, "group_size": 64, "ratio": 0.8},
     "stablelm-epoch-3b-preview": {"bits": 4, "sym": True, "group_size": 64, "ratio": 0.8},
-    "stable-zephyr-3b-dpo": {"bits": 4, "sym": False, "group_size": 64, "ratio": 0.8},
+    "stabilityai/stable-zephyr-3b-dpo": {"bits": 4, "sym": False, "group_size": 128, "ratio": 1.0},
```
Comment (echarlaix) [inline optimum/intel/openvino/configuration.py:54]: should it be ?
```suggestion
    "stabilityai/stablelm-zephyr-3b": {"bits": 4, "sym": False, "group_size": 128, "ratio": 1.0},
```

Comment (echarlaix) [reply optimum/intel/openvino/configuration.py:54]: cc @KodiaqQ what do you think ?

Comment (KodiaqQ) [reply optimum/intel/openvino/configuration.py:54]: @echarlaix, you're right, thanks!

**Pull request merged**